### PR TITLE
Get query for sort by and order

### PR DESCRIPTION
**Pull request by CRJenner**
Task 11 to order and sort query, with test. Error handling for a task with invalid order and task with invalid sort by

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -275,4 +303,26 @@ describe("Error handling", () => {
...
+      return request(app)
+        .get("/api/reviews?order=invalidOrder")
+        .expect(400)
+        .then(({ body }) => {
+          const { msg } = body;
+          expect(msg).toBe("Invalid order query, try again");
+        });
+    });
+  });
 });
```
Comment (elijaharley) [inline __tests__/app.test.js:386]: Please make sure you have tests for all of the the following queries  including topics - 
  - [x] Status 200, accepts `order` query, e.g. `?order=desc`
  - [x] Status 200, accepts `topic` query, e.g. `?topic=coding`
  - [ ] Status 404. non-existent `topic` query, e.g. `?topic=bananas`
  - [x] Status 200. valid `topic` query, but has no articles responds with an empty array of articles, e.g. `?topic=paper`

File: app.js [hunk 2]
```diff
@@ -33,8 +33,11 @@ app.use((err, request, response, next) => {
     response.status(400).send({ msg: "Invalid input, use a number" });
   } else if (err.code === "23503") {
     response.status(404).send({ msg: "error: User not found" });
+  } else if (err.code === "42703") {
+    response.status(400).send({ msg: "Invalid sort query, try again." });
+  } else if (err.code === "42601") {
+    response.status(400).send({ msg: "Invalid order query, try again" });
   }
```
Comment (elijaharley) [inline app.js:40]: Nice handling of different PSQL errors here

File: models/models.js [hunk 3]
```diff
@@ -55,7 +55,21 @@ exports.updateReviews = (review_id, inc_votes) => {
...
+    "title",
+    "designer",
+    "owner",
+    "review_body",
+    "category",
+    "votes",
+    "created_at",
+  ];
+  const orderQuery = ["asc", "desc"];
   let intitialQuery = `SELECT reviews.*,
```
Comment (elijaharley) [inline models/models.js:76]: Good greenlisting here

**Review by elijaharley — CHANGES_REQUESTED** [review]
Some good work here - make sure that you have tested for queries for categories here also - I have left a list for you to look at. You may need to add a checkExists utils function to do this which you can find an example of in the notes.

File: __tests__/app.test.js (hunk 1 above)
Comment (CRJenner) [reply __tests__/app.test.js:386]: For topics? Is that meant as category? I cannot find topic anywhere in the db.

**Pull request merged**